rocket_learn/rollout_generator/redis/redis_rollout_worker.py

```diff
@@ -80,8 +81,17 @@ def __init__(self, redis: Redis, name: str, match: Match,
         self.send_obs = send_obs
         self.dynamic_gm = dynamic_gm
         self.gamemode_weights = gamemode_weights
-        if self.gamemode_weights is not None:
-            assert sum(self.gamemode_weights.values()) == 1, "gamemode_weights must sum to 1"
+        if self.gamemode_weights is None:
+            self.gamemode_weights = {'1v1': 1/3, '2v2': 1/3, '3v3': 1/3}
+        assert sum(self.gamemode_weights.values()) == 1, "gamemode_weights must sum to 1"
+        self.target_weights = copy.copy(self.gamemode_weights)
+        # change weights from percentage of experience desired to percentage of gamemodes necessary (approx)
+        for k in self.gamemode_weights.keys():
+            b, o = k.split("v")
+            self.gamemode_weights[k] /= int(b)
+        weights_sum = sum(self.gamemode_weights.values())
+        self.gamemode_weights = {k: self.gamemode_weights[k] / weights_sum for k in self.gamemode_weights.keys()}
```

Comment:
I don't think it's a good idea to reuse a public attribute named as a constructor arg to store values different from the args provided, in other words, i think `self.game_mode` should be readonly for the sake of clarity and consistency